gui-test/framework/tests/racket.rkt

```diff
@@ -611,6 +621,45 @@
                              ; test that "def" remains selected afterwards...
                              [" \"abc\" "   "\"def\""    " \"ghi\" "]))
 
+(test-parens-behavior/full 'delete-empty-block-comment
+                           " #|" "" "|#"
+                           #\backspace
+                           '([" #" "|#"]
+                             [" #" ""]))
+(test-parens-behavior/full 'delete-bars-with-hash
+                           " |" "" "|#"
+                           #\backspace
+                           '([" " "|#"]
+                             [" " "#"]))
+(test-parens-behavior/full 'delete-one-bar-between-hashes-in-string
+                           " \"#|" "" "|#\""
+                           #\backspace
+                           '([" \"#" "|#\""]
+                             [" \"#" "|#\""]))
+(test-parens-behavior/full 'delete-escaped-double-quote-in-string
+                           "\"abcd \\\"" "" "\""
+                           #\backspace
+                           '(["\"abcd \\" "\""]
+                             ["\"abcd \\" "\""])) ; don't delete the non-escaped double quote
+
+;; test that backspace only removes one character in most cases in non-empty strings and comments
+(for ([open '("(" "[" "{" "\"" "|")]
+      [close '(")" "]" "}" "\"" "|")])
+  (define single-delete-scenarios
+    ;    description     before-cursor    after-cursor
+    '(("in-line-comment"  ";; abc def "    " ghi ")
+      ("in-block-comment" "#| abc def "  " ghi |#")
+      ("in-string"        "\" abc def "  " ghi \"")
+      ))
+  (for ([s single-delete-scenarios]
+        #:unless (and (string=? "\"" open)
+                      (string=? "in-string" (car s))))
+    (let* ([before (cadr s)]
+           [after (string-append close (caddr s))]
+           [before-and-open (string-append before open)]
+           [result (list before after)])
+      (test-parens-behavior/full (format "~a-~a" (string-append open close) (car s))
+                                 before-and-open "" after #\backspace `(,result ,result)))))
 
```

Comment:
need some tests for the new method.

Reply:
Are the tests in color-text.rkt suitable if I move them over, but applied to an instance of racket:text% instead of color:text%?  Should I delete them from color-text.rkt, or is there somewhere else we have tests for (not racket-specific) color:text<%>?